File: Activities/Activity 5/Activity5.cpp
/*
 * Activity 5: Expressions and formatted I/O
 * Group 5
 * Andriy Konyshev
 * Grant Loofbourrow
 * Nathan Saugstad
 */

#include <cmath> // for pow function
#include <iomanip> // for formatting output
#include <iostream> // for user input/output

using namespace std;

int main() {

    // Prompt for annual interest rate, Rate * 12
    double monthlyRate, loanAmount, numberOfMonths, monthlyPayment, totalAmountPaid;
    cout << "Enter an annual interest rate: ";
    cin >> monthlyRate;
    monthlyRate /= 1200.0;

    // Prompt for loan amount, L
    cout << "Enter the loan amount: ";
    cin >> loanAmount;

    // Prompt for number of monthly payments, N
    cout << "Over how many months will you pay this loan? ";
    cin >> numberOfMonths;

    // Payment = (Rate * pow(1 + Rate, N)) / (pow(1 + Rate, N) - 1) * L
    monthlyPayment = (monthlyRate * pow(1 + monthlyRate, static_cast<double>(numberOfMonths)))
         / (pow(1 + monthlyRate, static_cast<double>(numberOfMonths)) - 1) * loanAmount;

    // Discard trailing decimal digits
    monthlyPayment = static_cast<int>(monthlyPayment * 100) / 100.0;

    totalAmountPaid = monthlyPayment * numberOfMonths;

    // Display formatted output
    // Right-aligned labels
    // Values aligned by digit magnitudes
    cout << fixed  << showpoint << setprecision(2);
    cout << setw(24) << "Loan Amount:" << "  $ " << setw(8) << loanAmount << '\n'
        << setw(24) << "Monthly Interest Rate:" << "    " << setw(8) << monthlyRate * 100 << "%\n"
        << setw(24) << "Number of Payments:" << "    " << setw(5) << static_cast<int>(numberOfMonths) << '\n'
        << setw(24) << "Monthly Payment:" << "  $ " << setw(8) << monthlyPayment << '\n'
        << setw(24) << "Amount Paid Back:" << "  $ " << setw(8) << totalAmountPaid << '\n'
        << setw(24) << "Interest Paid:" <<  "  $ " << setw(8) << totalAmountPaid - loanAmount << endl;

    return 0;

}

File: Activities/Activity 5/Activity5.java
import java.util.Scanner;

public class Activity5 {

	public static void main(String[] args) {
		
		double monthlyRate, loanAmount, monthlyPayment, totalAmountPaid;
		int numberOfMonths;
		Scanner s = new Scanner(System.in);
		
		System.out.print("Enter an annual interest rate: ");
		monthlyRate = s.nextDouble() / 1200;
		System.out.print("Enter the loan amount: ");
		loanAmount = s.nextDouble();
		System.out.print("Over how many months will you pay this loan? ");
		numberOfMonths = s.nextInt();
		
		s.close();
		
		monthlyPayment = (monthlyRate * Math.pow(1 + monthlyRate, numberOfMonths)) 
				/ (Math.pow(1 + monthlyRate, numberOfMonths) - 1) * loanAmount;
		monthlyPayment = (int)(monthlyPayment * 100) / 100.0;
		
		totalAmountPaid = monthlyPayment * numberOfMonths;
		
		System.out.printf("%24s  $ %8.2f%n", "Loan Amount:", loanAmount);
		System.out.printf("%24s    %8.2f%n", "Monthly Interest Rate:", monthlyRate * 100);
		System.out.printf("%24s    %5d%n", "Number of Payments:", numberOfMonths);
		System.out.printf("%24s  $ %8.2f%n", "Monthly Payment:", monthlyPayment);
		System.out.printf("%24s  $ %8.2f%n", "Amount Paid Back:", totalAmountPaid);
		System.out.printf("%24s  $ %8.2f%n", "Interest Paid:", totalAmountPaid - loanAmount);
		
	}
	
}
